From 44d854a518f97cb65090420b0b9f55611a0ea932 Mon Sep 17 00:00:00 2001
From: Justin Ruggles <justin.ruggles@gmail.com>
Date: Mon, 22 Oct 2012 10:44:55 -0400
Subject: [PATCH 008/278] atrac3: return an error if extradata_size is not a
 specific known size

Also fixes 3 compiler warnings about using uninitialized variables.
---
 libavcodec/atrac3.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/atrac3.c b/libavcodec/atrac3.c
index 9cf9892..5bf992f 100644
--- a/libavcodec/atrac3.c
+++ b/libavcodec/atrac3.c
@@ -912,6 +912,7 @@ static av_cold int atrac3_decode_init(AVCodecContext *avctx)
     } else {
         av_log(NULL, AV_LOG_ERROR, "Unknown extradata size %d.\n",
                avctx->extradata_size);
+        return AVERROR(EINVAL);
     }
 
     /* Check the extradata */
-- 
1.7.9.5

